utils/git/clang_tidy_pr.py

```diff
@@ -0,0 +1,626 @@
+        type=str,
+        required=True,
+        help="Name of the repository containing the code",
+    )
+    parser.add_argument(
+        "--repository-root",
+        type=str,
+        required=True,
+        help="Path to the root of the repository containing the code",
+    )
+    parser.add_argument(
+        "--request-changes",
+        type=str,
+        required=True,
+        help="If 'true', then request changes if there are warnings, otherwise leave a comment",
+    )
+    parser.add_argument(
+        "--suggestions-per-comment",
+        type=int,
+        required=True,
+        help="Number of suggestions per comment",
+    )
+
+    args = parser.parse_args()
+
+    # The GitHub API token is sensitive information, pass it through the environment
+    github_token = os.environ.get("GITHUB_TOKEN")
+
+    github_api_url = os.environ.get("GITHUB_API_URL")
+    github_api_timeout = 10
+
+    warning_comment_prefix = (
+        "# :warning: `Clang-Tidy` found issue(s) with the introduced code"
+    )
+
+    diff_lines_per_file = get_diff_lines_per_file(
+        get_pull_request_files(
+            github_api_url,
+            github_token,
+            github_api_timeout,
+            args.repository,
+            args.pull_request_id,
+        )
+    )
+
+    with open(args.clang_tidy_fixes, encoding="utf_8") as file:
+        clang_tidy_fixes = yaml.safe_load(file)
+
+    if (
+        clang_tidy_fixes is None
+        or "Diagnostics" not in clang_tidy_fixes.keys()
+        or len(clang_tidy_fixes["Diagnostics"]) == 0
+    ):
+        print("No warnings found by Clang-Tidy")
+        dismiss_change_requests(
+            github_api_url,
+            github_token,
+            github_api_timeout,
+            args.repository,
+            args.pull_request_id,
+            warning_comment_prefix,
+        )
+        return 0
+
+    review_comments = list(
+        generate_review_comments(
+            clang_tidy_fixes, args.repository_root + "/", diff_lines_per_file
+        )
+    )
+
+    existing_pull_request_comments = list(
+        get_pull_request_comments(
+            github_api_url,
+            github_token,
+            github_api_timeout,
+            args.repository,
+            args.pull_request_id,
+        )
+    )
+
+    # Exclude already posted comments
+    for comment in existing_pull_request_comments:
+        review_comments = list(
+            filter(
+                lambda review_comment: not (
+                    review_comment["path"]
+                    == comment["path"]  # pylint: disable=cell-var-from-loop
+                    and review_comment["line"]
+                    == comment["line"]  # pylint: disable=cell-var-from-loop
+                    and review_comment["side"]
+                    == comment["side"]  # pylint: disable=cell-var-from-loop
+                    and review_comment["body"]
+                    == comment["body"]  # pylint: disable=cell-var-from-loop
```

Comment:
This reads awful. Can't you put the `pylint` hints over each line and have something like:
```
X == A
and Y == B
and Z == C
```
It would certainly read better. In fact, since it's the same warning to disable, can't you just put it right after the `for` line and disable the warning for the whole block?

Reply:
I didn't [write this](https://github.com/platisd/clang-tidy-pr-comments/blob/121c8f88ff614d196bbdfb15b641417423ec162f/run_action.py) so I don't really want to fix it (especially since pylint gives it a 10/10 score :p).

Reply:
As much as I appreciate them, I don't let auto-linters have the last word. After all, the whole point of having a linter is to make the code easier to read _for people_ ;-)

In any case, if this has been copied from somewhere else, then don't give it a second thought.